flux-refineck/src/checker.rs

```diff
@@ -923,6 +927,37 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
     }
 }
 
+fn type_of_shift<'a>(
+    num_ty: &'a Ty,
+    shift_by_ty: &'a Ty,
+    op: mir::BinOp,
+) -> (BaseTy, &'a rty::RefineArg, &'a rty::RefineArg, sigs::Sig<2>) {
+    match (num_ty.kind(), shift_by_ty.kind()) {
+        (Int!(num_base_ty, num_idxs), Int!(_, shift_by_idxs) | Uint!(_, shift_by_idxs)) => {
```

Comment:
We call this an `int_ty`.

```suggestion
        (Int!(num_int_ty, num_idxs), Int!(_, shift_by_idxs) | Uint!(_, shift_by_idxs)) => {
```